src/Data/Filters/RecordingsFilter.php

```diff
@@ -0,0 +1,50 @@
+<?php declare(strict_types=1);
+/**
+ * PHP Version 7
+ *
+ * @category   Data
+ * @package    Main
+ * @subpackage Images
+ * @author     Cecile <cecile.madjar@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+namespace LORIS\Data\Filters;
+
+use \Loris\Data\DataInstance;
+use \LORIS\Data\Filter;
+
+/**
+ * This is meant to be used by a provisioner to filter-out electrophysiology
+ * recordings when the user should not see them.
+ *
+ * @category   Data
+ * @package    Main
+ * @subpackage Images
+ * @author     Cecile <cecile.madjar@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class RecordingsFilter implements Filter
```

Comment:
There is now HasAnyPermissionOrUserSiteMatch filter if you want to remove one file from this PR :)

Reply:
actually, I tried it but this filter applies only to candidate list.

For a user to be able to see recordings, it needs to either have:
- permission to view all site EEG
- or permission to view own site EEG and the data belongs to the user's site

Same behaviour as the imaging basically (without the phantom headache ;)).

Reply:
HasAnyPermissionOrUserSiteMatch takes a list of permission to check before fallingback to `the data belongs to the user's site`.
so:
```
$filter = new HasAnyPermissionOrUserSiteMatch(
    array('permission to view all site EEG')
);
```
would filter the data according to the criteria explained in your comment.
And for 'permission to view own site EEG', this shoudl be covered by the hasAccess function.

Reply:
👍 
@xlecours  all done in the last commit.